### PR TITLE
Make /crate/:crate/latest/builds.json not strip .json suffix

### LINKED ISSUE #170: /crate/:crate/latest/builds.json strips the .json suffix
This may be useful for external badge services like [shields.io](https://shields.io/).

See also: https://github.com/badges/shields/issues/1382

### PR DESCRIPTION
**Pull request by coolshaurya**
Fixes #170 .

### CODE REVIEW

File: src/web/builds.rs [hunk 1]
```diff
@@ -43,18 +43,22 @@ pub fn build_list_handler(req: &mut Request) -> IronResult<Response> {
     let mut conn = extension!(req, Pool).get()?;
     let limits = ctry!(req, Limits::for_crate(&mut conn, name));
 
+    let is_json = req.url.path().join("/").ends_with(".json");
```
Comment (jyn514) [inline src/web/builds.rs:46]: nit: this allocates on every request
```suggestion
    let is_json = req.url.path().last().map_or(false, |segment| segment.ends_with(".json"));
```
https://docs.rs/iron/0.6.1/iron/struct.Url.html#method.path
I do think this is a slight change in behavior in that `/builds.json/` will no longer work, but that seems like a misfeature anyway.

**Comment by jyn514** [comment]
@syphar the "I'm feeling lucky" has been failing intermittently, do you have time to fix it? I think the issue is we have a much lower percentage of crates with over 100 stars in the test database, you could make it configurable by moving 500 into a query parameter and passing in a different number in tests.

**Comment by syphar** [comment]
@jyn514 I have time and can figure something out.

**Comment by syphar** [comment]
(will do a separate PR)

**Comment by jyn514** [comment]
Can you add a test for the new behavior? You can see examples at the bottom of src/web/mod.rs./

**Comment by jyn514** [comment]
Looks great, thanks!

**Pull request merged**